scripts/helpers/restyle-diff.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+#
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Use git and restyle-path to help reformat anything that differs from
+#  some base edit point.  Use before pushing a PR to make sure everything
+#  you've written is kosher to CI
+#
+# Usage:
+#  restyle-diff.sh [ref]
+#
+# if unspecified, ref defaults to "master"
+#
+
+here=${0%/*}
+
+set -e
+
+CHIP_ROOT=$(cd "$here/../.." && pwd)
+
+restyle-paths() {
+    url=https://github.com/restyled-io/restyler/raw/master/bin/restyle-path
+
+    sh <(curl --location "$url" -sSf) "$@"
```

Comment:
This "grab something from the internet and run it as a shell script" bit gives me the willies.  I guess at least on Mac `curl` will do useful cert verification against the OS cert store, so this is safe-ish against MITM there as long as all the trust roots in the cert store behave.  Is that also true on other OSes?

Also, this means that you have to be online to run the restyler.  I know you still need to be the first time you do because it's going to fetch docker stuff, of course...

How annoying would it be to just add `https://github.com/restyled-io/restyler` as a git submodule and then call it from the CHIP tree instead of doing this on every invocation?

Reply:
if we named the commit instead of pulling from master that would be equivalent, right?

Reply:
It would still have the "needs network access on every restyle invocation" issue, I think?  As opposed to only when you`git submodule update`

Reply:
further bullet-proofing can be done by constraining the protocol to https, disabling redirects, etc., but this is closing barn door stuff.   we really shouldn't ever be building open source outside some container, right?

Reply:
> This "grab something from the internet and run it as a shell script" bit gives me the willies. I guess at least on Mac `curl` will do useful cert verification against the OS cert store

and the user's exceptions, via openssl or whatevs

> so this is safe-ish against MITM there as long as all the trust roots in the cert store behave. 

assuming we trust github and restyled-io 

> Is that also true on other OSes?

yep

> Also, this means that you have to be online to run the restyler. I know you still need to be the first time you do because it's going to fetch docker stuff, of course...
> 
> How annoying would it be to just add `https://github.com/restyled-io/restyler` as a git submodule and then call it from the CHIP tree instead of doing this on every invocation?

I could just copy the script, but thought this was cuter

Reply:
updated to be more strict